### PR TITLE
SQL: [Tests] Re-enable testDriverConfigurationWithSSLInURL test with more logging

### PR DESCRIPTION
**Pull request by astefan**
Following https://github.com/elastic/elasticsearch/issues/41557, this PR re-enables `JdbcConfigurationTests.testDriverConfigurationWithSSLInURL` with more logging coming from the DriverManager hopefully getting more insight for the next time this test fails.

### CODE REVIEW

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample

**Comment by astefan** [comment]
@elasticmachine run elasticsearch-ci/2

File: x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java [hunk 1]
```diff
@@ -252,14 +256,20 @@ public void testSSLPropertiesOverride() throws Exception {
         assertSslConfig(props, JdbcConfiguration.create("jdbc:es://test?" + sslUrlProps.toString(), props, 0).sslConfig());
     }
     
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/41557")
-    public void testDriverConfigurationWithSSLInURL() {
+    @SuppressForbidden(reason = "JDBC drivers allows logging to Sys.out")
+    public void testDriverConfigurationWithSSLInURL() throws Exception {
```
Comment (matriv) [inline x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java]: I think `throws Exception` is not necessary.

**Review by matriv — APPROVED** [review]
LGTM

**Pull request merged**